storage/src/lib.rs

```diff
@@ -52,13 +58,28 @@ impl From<PathBuf> for StorageMode {
     }
 }
 
+impl From<Arc<TempDir>> for StorageMode {
+    fn from(tempdir: Arc<TempDir>) -> Self {
+        StorageMode::Test(Some(tempdir))
+    }
+}
+
+impl From<Option<Arc<TempDir>>> for StorageMode {
```

Comment:
Ideally there's a `new_test_mode` or something, would be clearer and 0 chance of leaking with previous setup.